.github/workflows/build-and-test.yml

```diff
@@ -504,8 +504,23 @@ jobs:
         source env/activate
         cmake --build ${{ steps.strings.outputs.build-output-dir }} -- explorer
 
+    - name: Build ttrt
+      shell: bash
+      run: |
+        source env/activate
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} -- ttrt
+
+    - name: Generate tests
+      shell: bash
+      run: |
+        source env/activate
+        ttrt query --save-artifacts
+        export SYSTEM_DESC_PATH="${GITHUB_WORKSPACE}/ttrt-artifacts/system_desc.ttsys"
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} -- check-ttmlir
```

Comment:
Hmm, could this cause a bit of confusion in the CI since we are rerunning the same tests from another job? If someone makes a mistake in FileCheck they will see the error in explorer job as well, which might not be what we want.

Reply:
Not sure about this one. @nobradovictt Thoughts?

Reply:
Maybe we could have a dedicated subdir for explorer tests that are still underneath the `/tests` root dir?

```bash
llvm-lit build/test/ttmlir/Explorer
```

Reply:
@vprajapati-tt suggested that we move explorer tests to `/tests` and we agreed that makes sense.

Reply:
My thoughts on this are:

We should have explorer specific tests, and it would be best if they reside in `/tests/` so we get coverage on if dialect changes are incompatible with the tests that we've written.

It's also important for us to have coverage for the explorer parsing as well. While our tests can test the capability of the tool and target edge cases, the tool is meant to be generalized for `tt-mlir` dialects. As such, holding increased coverage from these tests will provide exposure to the flaws of how models are being parsed in explorer (ex: a NamedAttribute called `shape` can have different underlying types, and this was causing a problem with the llama tests).

I think there is much to gain with autogenerating since even unrelated changes can cause problems in the explorer parsing engine, and it would be best for us to notice that without having to manually test these cases / write them in.

Reply:
Ok, let's try this out provided we do the following:
1. We don't use check-ttmlir within the explorer job in order to not have explorer CI red when FileCheck is failing.
2. The job is not the slowest one on CI.